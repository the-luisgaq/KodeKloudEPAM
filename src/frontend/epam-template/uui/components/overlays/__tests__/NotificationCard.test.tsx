import React from 'react';
import { renderSnapshotWithContextAsync, SvgMock } from '@epam/uui-test-utils';
import {
    ClearNotification, ErrorNotification, HintNotification, NotificationCard, SuccessNotification, WarningNotification,
} from '../NotificationCard';

describe('NotificationCard', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<NotificationCard id={ 1 } key="test" color="error" onClose={ jest.fn } onSuccess={ jest.fn } />);
        expect(tree).toMatchSnapshot();
    });

    it('should render with maximum props', async () => {
        const tree = await renderSnapshotWithContextAsync(<NotificationCard icon={ SvgMock } id={ 1 } key="test" color="error" onClose={ jest.fn } onSuccess={ jest.fn } />);
        expect(tree).toMatchSnapshot();
    });
});

describe('WarningNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<WarningNotification id={ 1 } key="test" onClose={ jest.fn } onSuccess={ jest.fn } />);
        expect(tree).toMatchSnapshot();
    });
});

describe('SuccessNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<SuccessNotification id={ 1 } key="test" onClose={ jest.fn } onSuccess={ jest.fn } />);
        expect(tree).toMatchSnapshot();
    });
});

describe('HintNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<HintNotification id={ 1 } key="test" onClose={ jest.fn } onSuccess={ jest.fn } />);
        expect(tree).toMatchSnapshot();
    });
});

describe('ErrorNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<ErrorNotification id={ 1 } key="test" onClose={ jest.fn } onSuccess={ jest.fn } />);
        expect(tree).toMatchSnapshot();
    });
});

describe('ClearNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<ClearNotification key="test" />);
        expect(tree).toMatchSnapshot();
    });
});
