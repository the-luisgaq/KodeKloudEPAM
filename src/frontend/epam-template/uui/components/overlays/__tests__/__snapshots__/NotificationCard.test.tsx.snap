// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearNotification should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="clearNotifications uui-clear-notifications"
  >
    <button
      class="uui-button-box uui-enabled -clickable uui-link_button root uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary"
      tabindex="0"
      type="button"
    >
      <div
        class="uui-caption uui-link-button-weight-semibold"
      >
        CLOSE ALL NOTIFICATIONS
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`ErrorNotification should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="uui-notification_card uui-color-error root"
    role="alert"
  >
    <div
      class="mainPath"
    >
      <div
        class="iconWrapper"
      >
        <div
          class="container uui-icon uui-enabled"
        >
          <svg
            class=""
            height="24"
            width="24"
          />
        </div>
      </div>
      <div
        class="content"
      />
      <div
        class="closeWrapper uui-notification_card-close_icon"
      >
        <button
          class="uui-button-box uui-enabled -clickable container uui-icon_button uui-color-neutral root closeIcon"
          tabindex="0"
          type="button"
        >
          <div
            class="container uui-icon uui-enabled"
          >
            <svg
              class=""
            />
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HintNotification should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="uui-notification_card uui-color-info root"
    role="alert"
  >
    <div
      class="mainPath"
    >
      <div
        class="iconWrapper"
      >
        <div
          class="container uui-icon uui-enabled"
        >
          <svg
            class=""
            height="24"
            width="24"
          />
        </div>
      </div>
      <div
        class="content"
      />
      <div
        class="closeWrapper uui-notification_card-close_icon"
      >
        <button
          class="uui-button-box uui-enabled -clickable container uui-icon_button uui-color-neutral root closeIcon"
          tabindex="0"
          type="button"
        >
          <div
            class="container uui-icon uui-enabled"
          >
            <svg
              class=""
            />
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationCard should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="uui-notification_card uui-color-error root"
    role="alert"
  >
    <div
      class="mainPath"
    >
      <div
        class="content"
      />
      <div
        class="closeWrapper uui-notification_card-close_icon"
      >
        <button
          class="uui-button-box uui-enabled -clickable container uui-icon_button uui-color-neutral root closeIcon"
          tabindex="0"
          type="button"
        >
          <div
            class="container uui-icon uui-enabled"
          >
            <svg
              class=""
            />
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationCard should render with maximum props 1`] = `
<DocumentFragment>
  <div
    class="uui-notification_card uui-color-error root"
    role="alert"
  >
    <div
      class="mainPath"
    >
      <div
        class="iconWrapper"
      >
        <div
          class="container uui-icon uui-enabled"
        >
          <svg
            class=""
            height="24"
            width="24"
          />
        </div>
      </div>
      <div
        class="content"
      />
      <div
        class="closeWrapper uui-notification_card-close_icon"
      >
        <button
          class="uui-button-box uui-enabled -clickable container uui-icon_button uui-color-neutral root closeIcon"
          tabindex="0"
          type="button"
        >
          <div
            class="container uui-icon uui-enabled"
          >
            <svg
              class=""
            />
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SuccessNotification should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="uui-notification_card uui-color-success root"
    role="alert"
  >
    <div
      class="mainPath"
    >
      <div
        class="iconWrapper"
      >
        <div
          class="container uui-icon uui-enabled"
        >
          <svg
            class=""
            height="24"
            width="24"
          />
        </div>
      </div>
      <div
        class="content"
      />
      <div
        class="closeWrapper uui-notification_card-close_icon"
      >
        <button
          class="uui-button-box uui-enabled -clickable container uui-icon_button uui-color-neutral root closeIcon"
          tabindex="0"
          type="button"
        >
          <div
            class="container uui-icon uui-enabled"
          >
            <svg
              class=""
            />
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WarningNotification should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="uui-notification_card uui-color-warning root"
    role="alert"
  >
    <div
      class="mainPath"
    >
      <div
        class="iconWrapper"
      >
        <div
          class="container uui-icon uui-enabled"
        >
          <svg
            class=""
            height="24"
            width="24"
          />
        </div>
      </div>
      <div
        class="content"
      />
      <div
        class="closeWrapper uui-notification_card-close_icon"
      >
        <button
          class="uui-button-box uui-enabled -clickable container uui-icon_button uui-color-neutral root closeIcon"
          tabindex="0"
          type="button"
        >
          <div
            class="container uui-icon uui-enabled"
          >
            <svg
              class=""
            />
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
